consensus/proto_array/src/proto_array.rs

```diff
@@ -142,11 +126,33 @@ impl ProtoArray {
                 continue;
             }
 
-            let node_delta = deltas
+            let mut node_delta = deltas
                 .get(node_index)
                 .copied()
                 .ok_or(Error::InvalidNodeDelta(node_index))?;
 
+            // If we find the node for which the proposer boost was previously applied, decrease
+            // the delta by the previous score amount.
+            if self.previous_proposer_boost.root == node.root {
```

Comment:
```suggestion
            if self.previous_proposer_boost.root != Hash256::zero()
                && self.previous_proposer_boost.root == node.root
            {
```

I'm not *really* sure this is necessary, but it prevents the scenario where a block with root `Hash256::zero()` will have its delta reduced without ever being increased (since the path to *increase* the boost is gated by `!= Hash256::zero()`).

It shouldn't happen in production, however we might see it during artificial testing. Also the spec likes to alias genesis to `Hash256::zero()` so perhaps this is useful defensively.